pkg/iter/entry_iterator.go

```diff
@@ -403,7 +402,6 @@ func (i *nonOverlappingIterator) Next() bool {
 		if i.curr != nil {
 			i.curr.Close()
```

Comment:
Why are we closing the iterator here?

Reply:
Because we're done with it.

Reply:
Ah. I misread `for i.curr == nil || !i.curr.Next() { }` and though we were not done iterating. Thanks!